### PR TITLE
[PR] General updates

### PR DESCRIPTION
**Pull request by jeremyBass**
## Changes
* Updates the tracker to use the Universal Tracker from Google
* provides a choice on how the site is to be segmented via a series of dropdown options 
* make it optional to use a theme level event file
* allow the theme level events to override or extend the defaults
* provide the same event handling options for jQuery UI since our spine has them

## Left to do

- [x] Provide the Spine Framework default events (should be optional i'd guess?  maybe not)
- [x] fill out the WSU organizational schema
- [x] Replace the "some label" in the media element events
- [x] Replace hard coded WSUWP Platform app level ID with filter we can hook into.

### CODE REVIEW

File: js/mediaelement-events.js [hunk 1]
```diff
@@ -16,19 +16,23 @@ _wpmejsSettings.success = function( mejs ) {
...
 
 		if(typeof jQuery.jtrack !=="undefined"){
+			/* we can do so much better then this, but it'll work for now
+				just update to do more robust tracking as we can.
+			*/
+			
 			// Event listener for when the video starts playing
 			mejs.addEventListener( 'playing', function( e ) {
-				jQuery.jtrack.trackEvent(pageTracker,"Audio", 'playing');
+				jQuery('body').trackEvent ('Audio', 'playing', "some lable", 1 );
```
Comment (jeremyfelt) [inline js/mediaelement-events.js]: What should take the place of "some label" here? Also s/lable/label/

Comment (jeremyBass) [inline js/mediaelement-events.js]: oh yes that should be the time I'd guess.. that is a little bit of choice there, but I think the time of the action is the most logical

**Comment by jeremyfelt** [comment]
From the perspective of reducing the number of options:

* It'd be nicer to defer to the theme automatically rather than having an option for **Extend Defaults**
* For **Use jQuery UI**, what are the benefits/drawbacks of having it enabled/disabled? Can we just default to on and not present the option?
* I'd like to ensure the **Turn on debug** option only appears in local development.

We're probably best creating a Settings -> Analytics page at this point to reduce how overwhelming the General settings page is.

**Comment by jeremyfelt** [comment]
Where is the work being done on the Campus / College / Parent Unit / Unit break down?

I'd like to add that as another taxonomy to the [University Taxonomy](https://github.com/washingtonstateuniversity/WSU-University-Taxonomy) repository as it has purposes other than analytics as well. I know we plan on using this data for the people directory as well.

We can then auto fill these drop downs based on what is in the taxonomy.

**Comment by jeremyBass** [comment]
> •It'd be nicer to defer to the theme automatically rather than having an option for Extend Defaults

The reason I did it this way is as a user I wanted all of the defaults, but I needed to add just one more rule.  

> •For Use jQuery UI, what are the benefits/drawbacks of having it enabled/disabled? Can we just default to on and not present the option?

If you are not using any of the jQuery UI parts (out side of the Spine) then there is no need to load rules for it.  The goal is to load as few rules as we can

> •I'd like to ensure the Turn on debug option only appears in local development.

It's there to override that local check you had, so makes sense


> We're probably best creating a Settings -> Analytics page 

:+1:

**Comment by jeremyBass** [comment]
> Where is the work being done on the Campus / College / Parent Unit / Unit break down?

Bob is doing that, and yes agreed it was intended to have it in a central place.

**Comment by jeremyfelt** [comment]
<blockquote>The reason I did it this way is as a user I wanted all of the defaults, but I needed to add just one more rule.</blockquote>

A theme file can extend the default file, right?

<blockquote>If you are not using any of the jQuery UI parts (out side of the Spine) then there is no need to load rules for it. The goal is to load as few rules as we can</blockquote>

I'm not familiar with the jQuery UI rules. What events are being toggled with this setting?

**Comment by jeremyfelt** [comment]
Chatted with @jeremyBass in person.

* We're going to provide an action for themes to hook in and replace or extend the object provided by `default_events.js`. This allows us to remove the **Extend Defaults** option.
* May need to reword the **Use jQuery UI** option to **Track jQuery UI** events. We don't care about these events globally, but somebody may want to track them at a site level. I'm personally still up in the air on this as another creative solution may be available. Will wait to see an example implementation.
* Bob is working on a unit taxonomy that we'll load into the WSU University Taxonomy so that we can pre-fill the campus, college, parent unit, unit boxes.

**Comment by jeremyfelt** [comment]
What should happen if GA is not entered for a site? Should we remove that from the tracker data entirely or continue to provide it, just with an empty string as the tracker ID?

I think now we do the latter.

**Comment by jeremyBass** [comment]
> What should happen if GA is not entered for a site?

The way it has been reset is yes if there is false (no site GA provided ) that we are still pushing it out.  It is not an empty string, it's "false".   The reason is that it auto corrects, and although is not firing the site scope GA, it is firing the Global, and the App scopes.

**Comment by jeremyfelt** [comment]
RE: empty site level GA ID

<blockquote>It is not an empty string, it's "false".</blockquote>

Ahh, yes - I missed that. So we're good the way it is now I believe.

**Comment by jeremyfelt** [comment]
* On/off flags are in for tracking global, app, and site analytics.
* A hard coded array of unit types is in.

Will hard code campuses and colleges in about an hour if it doesn't get hit first.

**Comment by jeremyfelt** [comment]
Colleges and campuses are in.

**Comment by jeremyfelt** [comment]
How are we looking @jeremyBass? I'm a little hesitant to ship on Friday, but as long as we do it in early afternoon so we have time to revert, I'm good.

I'd be comfortable with 8:30am (pre open lab) or 1:30pm (post lunch) if we're game tomorrow.

**Comment by jeremyBass** [comment]
@jeremyfelt , I'm not feeling all that well and I may call in tomorrow, and although I don't expect any issues, I think we should wait for Monday or Tuesday unless there is any thing pressing.  We are going to do a full test tomorrow morning with our whole unit on the uxtest.web.wsu.edu site, which has the plugin basically hard coded in, so we should be able to root out any issues there if there are any.  

I know even if I call in, I'll be online at some point mid day anyways so if we want to we can do the 1:30 for sure.  I'm comfortable either way.

**Comment by jeremyfelt** [comment]
I'm much more comfortable waiting until Monday, so :+1: on that plan if it goes that way. Hope you feel better!

**Comment by jeremyBass** [comment]
super sounds good and thanks I hope I do as well.  I'll try to test things out first thing in the morning regardless for a final check too.

**Comment by jeremyfelt** [comment]
Let's go for it.

**Pull request merged**